src/wp-includes/blocks/template-part.php

```diff
@@ -189,6 +189,11 @@ function build_template_part_block_area_variations() {
  * @return array Array containing the block variation objects.
  */
 function build_template_part_block_instance_variations() {
+	// Block themes are unavailable during installation.
+	if ( wp_installing() ) {
+		return;
```

Comment:
Should this be returning an empty array instead of implicitly null?

My first guess at the test failure is that PHP 8.0 is fatalling because of passing the returned null to `array_merge()` where this is called on line 242.

Reply:
> Should this be returning an empty array instead of implicitly null?

@anomiex Yep, that's the cause of the fatal.